python/graphstorm/dataloading/utils.py

```diff
@@ -113,20 +112,21 @@ def modify_fanout_for_target_etype(g, fanout, target_etypes):
         edge_fanout_lis.append(edge_fanout_dic)
     return edge_fanout_lis
 
-def flip_node_mask(dist_tensor):
+def flip_node_mask(dist_tensor, indices):
     """ Flip the node mask (0->1; 1->0) and return a flipped mask.
         This is equivalent to the `~` operator for boolean tensors.
 
         Parameters
         ----------
         dist_tensor: dgl.distributed.DistTensor
             The input mask
+        indices: torch.Tensor
+            The vector node IDs that belong to the local rank
 
         Returns
         -------
-        flipped mask: dgl.distributed.DistTensor
+        flipped mask: torch.Tensor
     """
-    flipped_dist_tensor = deepcopy(dist_tensor)
-    for idx in range(dist_tensor.shape[0]):
-        flipped_dist_tensor[idx] = 1 - dist_tensor[idx]
+    flipped_dist_tensor = th.ones(dist_tensor.shape, dtype=dist_tensor.dtype)
```

Comment:
this is not scalable. you should create a distributed tensor to store the flipped mask.